core/trino-main/src/main/java/io/trino/execution/SqlQueryExecution.java

```diff
@@ -379,12 +387,27 @@ public void start()
                     return;
                 }
 
+                planningThread = currentThread();
+                stateMachine.getStateChange(PLANNING).addListener(() -> {
+                    if (stateMachine.getQueryState() == FAILED) {
+                        synchronized (this) {
+                            if (planningThread != null) {
+                                planningThread.interrupt();
```

Comment:
How does this relate to the interrupt delivered here
https://github.com/trinodb/trino/blob/f13283bcc5a7507fa5396cdec9afbda93d642ee6/core/trino-main/src/main/java/io/trino/dispatcher/LocalDispatchQuery.java#L102 
?

Reply:
It is a different thread. `queryExecutionFuture.cancel(true)` interrupts thread started in `LocalDispatchQueryFactory::createDispatchQuery`.
The actual planning thread is started from that thread in `LocalDispatchQuery::startExecution`